org.eclipse.lsp4e/src/org/eclipse/lsp4e/operations/codeactions/LSPCodeActionQuickAssistProcessor.java

```diff
@@ -112,8 +111,8 @@ public boolean canAssist(IQuickAssistInvocationContext invocationContext) {
 		CodeActionParams params = prepareCodeActionParams(document, invocationContext.getOffset(), invocationContext.getLength());
 
 		try {
-			CompletableFuture<List<Either<Command, CodeAction>>> anyActions = executor.collectAll(ls -> ls.getTextDocumentService().codeAction(params)).thenApply(s -> s.stream().flatMap(List::stream).collect(Collectors.toList()));
-			if (anyActions.get(200, TimeUnit.MILLISECONDS).stream().filter(LSPCodeActionMarkerResolution::canPerform).collect(Collectors.toList()).isEmpty()) {
+			CompletableFuture<List<Either<Command, CodeAction>>> anyActions = executor.collectAll(ls -> ls.getTextDocumentService().codeAction(params)).thenApply(s -> s.stream().flatMap(List::stream).toList());
```

Comment:
this could be even a bit simpler, I have created https://github.com/eclipse/lsp4e/pull/663/files

Reply:
Thanks, honestly I haven't paid much attention to the streams but rather went and fixed all cases using Collectors.toList  after noticing it in one place.